### DISCUSSION TITLE
Importing modules (numpy )

### DISCUSSION THREAD

**Discussion by ghost** [IronLanguages/ironpython3#1642] [General]
Hi guys, is it possible to import modules with the latest ironPython version? I followed the steps in the README but still fails to load the modules in the python script.

**Comment by slozier** [comment]
It's still not possible to import C extension modules (numpy, pandas, etc). I keep hoping someone will revive the ironclad project (https://github.com/IronLanguages/ironclad) but I personally haven't had a chance to look at it.

**Reply by hesperaux** [reply]
Hi @slozier, new user to ironpython here. Attempting to use it with .net7.
Would you mind explaining how to use the nuget package and install python packages that are NOT cpython based? I am interested in using pyvisa-py and pyvisa, which I believe are not dependent on cpython.

**Reply by slozier** [reply]
@hesperaux I usually use the cli to install using pip and then copy the files over. For example `ipy -m pip install pyvisa-py==0.4.1 --user` and then grab the files from the install folder (e.g. `%AppData%\Roaming\Python\IronPython34\site-packages` on Windows). Alternatively you can grab the files directly from PyPI. You can then add the folder containing these packages to the search paths using `engine.SetSearchPaths` and you should then be able to import the packages.

**Reply by hesperaux** [reply]
Thank you for the fast response and the work you do on the project. I'll give this a try later today. If I have more questions I'll open another thread. Thanks again!

**Comment by BCSharp** [comment]
> I keep hoping someone will revive the ironclad project

I was planning to do just that after:

1. Finishing all outstanding changes from "What's new in 3.0" — in practice there are two main features left:
 * keyword-only/non-keyword parameters
 * full and efficient support of API taking bytes-like arguments
2. Making IPython work (this will require adding or completing several module IPython depends on).

Perhaps increasing test coverage of `ctypes` would be good before engaging `numpy` support, or maybe done in parallel with `ironclad`, as need arises.

All together, it can be a year-long work (in lead time).

**Reply by slozier** [reply]
Finally managed to get `ironclad` to build. I think it works, although I couldn't get `numpy` to import (don't think it ever worked with 2.7).

I just pushed a branch with a dockerized build environment which can hopefully serve as a starting point. If I find a bit of time I might try and "rebase" it to Python 3.4.
